jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsSettingsConfigurable.kt

```diff
@@ -0,0 +1,222 @@
+            cloudDebugExecutableTextField.emptyText.setText(autoDetectPath)
+        }
+        field.addBrowseFolderListener(
+            message("aws.settings.find.title", cliName),
+            message("aws.settings.find.description", cliName),
+            project,
+            FileChooserDescriptorFactory.createSingleLocalFileDescriptor()
+        )
+        return field
+    }
+
+    // modifyMessageBasedOnDetectionStatus will append "Auto-detected: ...." to the
+    // message if the executable is found this is used for setting the empty box text
+    private fun getSavedExecutablePath(executableType: ExecutableType<*>, modifyMessageBasedOnDetectionStatus: Boolean): String? = try {
+        ExecutableManager.getInstance().getExecutable(executableType).thenApply {
+            if (it is ExecutableWithPath) {
+                if (it !is ExecutableInstance.Executable) {
+                    return@thenApply (it as ExecutableWithPath).executablePath.toString()
+                } else {
+                    val path = it.executablePath.toString()
+                    val autoResolved = it.autoResolved
+                    if (autoResolved && modifyMessageBasedOnDetectionStatus) {
+                        return@thenApply message("aws.settings.auto_detect", path)
+                    } else if (autoResolved) {
+                        // If it is auto detected, we do not want to return text as the
+                        // box will be filled by empty text with the auto-resolve message
+                        return@thenApply null
+                    } else {
+                        return@thenApply path
+                    }
+                }
+            }
+            null
+        }.toCompletableFuture().join()
+    } catch (ignored: CompletionException) {
+        null
+    }
+
+    private fun validateAndSaveCliSettings(
+        textField: JBTextField,
+        executableName: String,
+        executableType: ExecutableType<*>,
+        saved: String?,
+        currentInput: String?
+    ) {
+        // If input is null, wipe out input and try to autodiscover
+        if (currentInput == null) {
+            ExecutableManager.getInstance().removeExecutable(executableType)
+            ExecutableManager.getInstance()
+                .getExecutable(executableType)
+                .thenRun {
+                    val autoDetectPath = getSavedExecutablePath(executableType, true)
+                    runInEdt(ModalityState.any()) {
+                        if (autoDetectPath != null) {
+                            textField.emptyText.setText(autoDetectPath)
+                        } else {
+                            textField.emptyText.setText("")
+                        }
+                    }
+                }
+            return
+        }
+        if (currentInput == saved) {
+            return
+        }
+        val path: Path
+        try {
+            path = Paths.get(currentInput)
```

Comment:
split declaration is a bit weird, but i think the flow still makes more sense like this

Reply:
Yeah I tried refactoring it but decided to leave it as is